### PR TITLE
Responsive sized Spacer components

### PR DESCRIPTION
**Pull request by jsteinb**
### Description

This PR adds sized responsive Spacer components (Spacer4, Spacer8, Spacer12, Spacer16, Spacer20, Spacer24, Spacer32, Spacer40, Spacer48, Spacer56, Spacer64, Spacer72, Spacer80) which have responsiveness built-in. These Spacer components align with the naming scheme that is currently being used on the design side in our Figma design system. [https://www.figma.com/file/P31na2HYB09sF7v5B0nqkJ/Sofar-Design-System?node-id=438%3A2](https://www.figma.com/file/P31na2HYB09sF7v5B0nqkJ/Sofar-Design-System?node-id=438%3A2)

### Motivation and Context

Our existing Spacer component is not responsive by default, and the only way to achieve responsive spacing is to manually give the component the responsive sizes for all screen sizes. This change will make it easier for developers to add responsive spacing and to have a common language around spacers between developers and design.

### Screenshots (if appropriate):

Screen Shot 2020-09-08 at 12 42 08 PM

### Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (non-breaking change which improves functionality or refactors code)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/atoms/Spacer/index.tsx [hunk 1]
```diff
@@ -1,7 +1,62 @@
...
+export const Spacer24: React.SFC = () => (
+  <Spacer mr={[4, 4, 5, 6, 6]} mb={[4, 4, 5, 6, 6]} />
+);
+
+export const Spacer32: React.SFC = () => (
+  <Spacer mr={[6, 6, 7, 8, 8]} mb={[6, 6, 7, 8, 8]} />
+);
+
+export const Spacer40: React.SFC = () => (
+  <Spacer mr={[5, 5, 6, 10, 10]} mb={[5, 5, 6, 10, 10]} />
```
Comment (bitscraps) [inline src/atoms/Spacer/index.tsx:41]: Can I just check this is the right way round with spacer 32.  The reason I ask is every other spacer is bigger than the previous but this one is smaller than the last on smaller devices

**Review by bitscraps — COMMENTED** [review]
Looks good just a comment about one of the spacers as the sizes look like they maybe incorrect but it maybe the case that it is how the design system is designed.

File: src/atoms/Spacer/index.tsx (hunk 1 above)
Comment (jsteinb) [reply src/atoms/Spacer/index.tsx:41]: good catch, I mixed something up there. Let me fix that

**Comment by jsteinb** [comment]
> Looks good just a comment about one of the spacers as the sizes look like they maybe incorrect but it maybe the case that it is how the design system is designed.

FYI I revised the sizes of the responsive spacers, there were a few sizes that did need to be adjusted. I think it should make more sense now. Overall the sizes should map roughly (not exactly) to the responsive behavior of typography elements like titles and headings.

**Review by bitscraps — APPROVED** [review]
All the number go up sequentially now so will approve as im assuming the number correspond to the design docs

**Review by liondedan — APPROVED** [review]

**Pull request merged**